### ISSUE TITLE
How to add “alt tag” to mautic image?

### ISSUE THREAD

**Issue by 1DayWebs** [mautic/mautic-wordpress#101]
My SEO plugin is flagging the mautic image because it has no alt tag ... can one be added?  Where?

**Comment by shulard** [comment]
Hello @1DayWebs !

Of course one can be added. We can update the `wpmautic_inject_noscript` function which defines the tracking image to inject a label in the `alt` attribute.

Do you want to work on a PR ?

**Comment by enterth** [comment]
Hello @shulard 

Is it possible to put "alt text" optionnal?

[https://i.imgur.com/hdjbVLW.png](url)

Because on firefox alt text is written in raw text on "mtracking.gif" while using "shortcode tag plugin", i had to remove alt="%s" in shortcodes.php to avoid this issue.

Thanks for this plugin btw,

Regards

**Comment by shulard** [comment]
Hello @enterth !

The alt text is using WordPress translation feature so you already can remove it (or update it) using the following code in your theme :

```php
add_filter( 'gettext', 'wpse_translate_string', 10, 3 );
function wpse_translate_string( $translation, $text, $domain ) {
    if ( 'wp-mautic' === $domain && 'Mautic Tags' === $text ) {
        $translation = 'Updated value.';
    }

    return $translation;
}
```

@1DayWebs you can already use the same code to define the alt text with the value you want.

However this alt tag is only added if you use the shortcode in your page like this :

```
[mautic type="tags" values="addtag,-removetag"]
```

We still need to update the "noscript" tag with the same alt text.

**Comment by shulard** [comment]
Hello @1DayWebs, @enterth !

I just published the `v2.4.1` release on GitHub and on WordPress.org. This release adds an alt text in the `<noscript>` tag that can be updated with the same code I gave you in my previous comment.

You can see the updated code in the PR #109.

I close this issue which is fixed for me now. Feel free to reopen it if required !
